substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1053,17 +1111,28 @@ impl<T: Config> Pallet<T> {
 	/// `Validators` or `VoterList` outside of this function is almost certainly
 	/// wrong.
 	pub fn do_remove_validator(who: &T::AccountId) -> bool {
-		let outcome = if Validators::<T>::contains_key(who) {
-			Validators::<T>::remove(who);
-			let _ = T::VoterList::on_remove(who).defensive();
-			true
-		} else {
-			false
+		let outcome = match Self::status(who) {
+			Ok(StakerStatus::Validator) => {
+				// make sure the validator is chilled before removing it.
+				let outcome = Self::do_chill_validator(who);
+				T::EventListeners::on_validator_remove(who);
+				outcome
+			},
+			Ok(StakerStatus::Idle) | Err(_) if T::TargetList::contains(who) => {
+				// try to remove "dangling" target. A dangling target does not have a bonded stash
+				// but is still part of the target list because a previously removed stash still has
+				// nominations from active nominators.
+				T::EventListeners::on_validator_remove(who);
+				false
+			},
+			_ => false,
```

Comment:
Just pointing out that this is a silent error for a nominator, we could probably have a `defensive` here instead.